add Particular.CodeRules 0.4.0

adamralph opened the pull request.

bording commented:
I'm not sure we want to add this package to this repo. This means we have to enforce our coding styles on the actual template content.

The templates are currently designed to match default VS as much as possible, which is why the using statements are located where they are, and we don't need to be enforcing `ConfigureAwait` in template code.

adamralph commented:
@bording I've only actually added the package to the tests project. Initially, I added it to all of them, and fixed the violations, but then I realised that we don't want the package ref to travel with the templates, so I removed it, but I left the violation fixes in.

However, you are correct that templates should remain as they are, so I'll revert the violation fixes.

bording commented on src/Tests/Tests.csproj:
```diff
@@ -9,6 +9,7 @@
     <PackageReference Include="NUnit" Version="3.12.0" />
     <PackageReference Include="NUnit3TestAdapter" Version="3.17.0" />
     <PackageReference Include="Particular.Approvals" Version="0.2.0" />
+    <PackageReference Include="Particular.CodeRules" Version="0.4.0" PrivateAssets="All" />
```
Nitpick: Test project doesn't need this to be `PrivateAssets="All"`

adamralph replied:
Good point. I'll change that.

adamralph commented:
@bording given that the only comment from your last review is addressed, I'm merging, but we can revisit if there's anything left outstanding.
